File: AdminPanel1/AdminPanel1.Web/Modules/Product/Product/ProductRow.cs
using AdminPanel1.SouqCom;
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System;
using System.ComponentModel;

namespace AdminPanel1.Product;

[ConnectionKey("souq"), Module("Product"), TableName("Product")]
[DisplayName("Product"), InstanceName("Product")]
[ReadPermission("Product")]
[ModifyPermission("Product")]
[ServiceLookupPermission("Product")]
[LookupScript]
public sealed class ProductRow : Row<ProductRow.RowFields>, IIdRow, INameRow
{
    const string jCat = nameof(jCat);

    [DisplayName("Id"), Identity, IdProperty]
    public int? Id { get => fields.Id[this]; set => fields.Id[this] = value; }

    [DisplayName("Name"), QuickSearch, NameProperty]
    public string Name { get => fields.Name[this]; set => fields.Name[this] = value; }

    [DisplayName("Description")]
    public string Description { get => fields.Description[this]; set => fields.Description[this] = value; }

    [DisplayName("Price"), Size(18)]
    public decimal? Price { get => fields.Price[this]; set => fields.Price[this] = value; }

    [DisplayName("Cat"), ForeignKey("Category", "Id"), LeftJoin(jCat), TextualField(nameof(CatName))]
    [LookupEditor(typeof(CategoryRow))]
    public int? CatId { get => fields.CatId[this]; set => fields.CatId[this] = value; }

    [DisplayName("Photo"),ImageUploadEditor]
    public string Photo { get => fields.Photo[this]; set => fields.Photo[this] = value; }

    [DisplayName("Type"), Column("type"), Size(50)]
    public string Type { get => fields.Type[this]; set => fields.Type[this] = value; }

    [DisplayName("Supplier Name"), Size(50)]
    public string SupplierName { get => fields.SupplierName[this]; set => fields.SupplierName[this] = value; }

    [DisplayName("Entry Date")]
    public DateTime? EntryDate { get => fields.EntryDate[this]; set => fields.EntryDate[this] = value; }

    [DisplayName("Review Url")]
    public string ReviewUrl { get => fields.ReviewUrl[this]; set => fields.ReviewUrl[this] = value; }

    [DisplayName("Quantity")]
    public int? Quantity { get => fields.Quantity[this]; set => fields.Quantity[this] = value; }

    [DisplayName("Cat Name"), Expression($"{jCat}.[Name]")]
    public string CatName { get => fields.CatName[this]; set => fields.CatName[this] = value; }

    public class RowFields : RowFieldsBase
    {
        public Int32Field Id;
        public StringField Name;
        public StringField Description;
        public DecimalField Price;
        public Int32Field CatId;
        public StringField Photo;
        public StringField Type;
        public StringField SupplierName;
        public DateTimeField EntryDate;
        public StringField ReviewUrl;
        public Int32Field Quantity;

        public StringField CatName;
    }
}

File: AdminPanel1/AdminPanel1.Web/Modules/ServerTypes/Product/ProductRow.ts
import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib";

export interface ProductRow {
    Id?: number;
    Name?: string;
    Description?: string;
    Price?: number;
    CatId?: number;
    Photo?: string;
    Type?: string;
    SupplierName?: string;
    EntryDate?: string;
    ReviewUrl?: string;
    Quantity?: number;
    CatName?: string;
}

export abstract class ProductRow {
    static readonly idProperty = 'Id';
    static readonly nameProperty = 'Name';
    static readonly localTextPrefix = 'Product.Product';
    static readonly lookupKey = 'Product.Product';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<ProductRow>('Product.Product') }
    static async getLookupAsync() { return getLookupAsync<ProductRow>('Product.Product') }

    static readonly deletePermission = 'Product';
    static readonly insertPermission = 'Product';
    static readonly readPermission = 'Product';
    static readonly updatePermission = 'Product';

    static readonly Fields = fieldsProxy<ProductRow>();
}